### DISCUSSION TITLE
mbedTLS handshake issue

### DISCUSSION THREAD

**Discussion by pmeskoni** [cesanta/mongoose#1536] [Ask questions, get answers] [answered]
When using MbedTLS, the handshake failed with a -76 error or sometimes a -75 error during the callbacks to mg_net_send() and mg_net_recv(). I believe this is because the logic to determine if Mongoose "should block" is not right.

in Mongoose 7.6 tls_mbed.c:

     static bool mg_wouldblock(int n) {
           return n < 0 &&
                      (errno == EINPROGRESS || errno == EAGAIN || errno == EWOULDBLOCK);
     }

We changed to this:

     static bool mg_wouldblock(int n) {
           return n < 0 &&
                      mg_sock_would_block();
     }

Has anyone else seen this issue?

Thanks,
PM

**Comment by cpq** [comment, marked as answer]
@pmeskoni could you check against the latest HEAD please?
